common/docker/docker.go

```diff
@@ -0,0 +1,622 @@
+package docker
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/charmbracelet/lipgloss"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/image"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/client"
+	"github.com/rotisserie/eris"
+
+	"pkg.world.dev/world-cli/common/config"
+	"pkg.world.dev/world-cli/common/logger"
+	"pkg.world.dev/world-cli/tea/style"
+)
+
+type BuildOutput struct {
+	Stream string `json:"stream"`
+	Aux    struct {
+		ID string `json:"ID"`
+	} `json:"aux"`
+}
+
+func Start(ctx context.Context, cfg *config.Config, configurations ...GetDockerConfig) error { //nolint:gocognit
+	cli, err := getCli()
```

Comment:
I think it's better to create a wrapper Docker Client struct so that you don't have to create and stop a new client everytime you want to run a docker command. Additionally, it will also remove a little bit of the headache of having to pass in `config.Config` again and again.

so something like this in `/docker/client.go`

```go
type Client struct {
	client.Client
	cfg config.Config
}
```

Then all the docker commands become a a method to `Client` struct.

You can also make it the responsibility of the caller (i.e. `/cmd/world/cardinal/start.go`) to call `defer client.Stop()` once.

Reply:
Also, this file is a bit too large so splitting the methods across different file would make sense.

The public methods like start/stop/purge can go to client.go. The private utility methods for volume and networking can go to `client_volume.go` and client_network.go` respectively.

Reply:
Created the wrapper Docker Client struct and separated the function to the respective file.